hub/core/compression.py

```diff
@@ -76,23 +104,95 @@ def decompress_array(buffer: Union[bytes, memoryview], shape: Tuple[int]) -> np.
 
     Args:
         buffer (bytes, memoryview): Buffer to be decompressed. It is assumed all meta information required to
-            decompress is contained within `buffer`.
-        shape (Tuple[int]): Desired shape of decompressed object. Reshape will attempt to match this shape before returning.
+            decompress is contained within `buffer`, except for byte compressions
+        shape (Tuple[int], Optional): Desired shape of decompressed object. Reshape will attempt to match this shape before returning.
+        dtype (str, Optional): Applicable only for byte compressions. Expected dtype of decompressed array.
+        compression (str, Optional): Applicable only for byte compressions. Compression used to compression the given buffer.
 
     Raises:
-        SampleDecompressionError: Right now only buffers compatible with `PIL` will be decompressed.
+        SampleDecompressionError: If decompression fails.
+        ValueError: If dtype and shape are not specified for byte compression.
 
     Returns:
         np.ndarray: Array from the decompressed buffer.
     """
-
+    if compression and get_compression_type(compression) == "byte":
+        if dtype is None or shape is None:
+            raise ValueError("dtype and shape must be specified for byte compressions.")
+        try:
+            decompressed_bytes = decompress_bytes(buffer, compression)
+            return np.frombuffer(decompressed_bytes, dtype=dtype).reshape(shape)
+        except Exception:
+            raise SampleDecompressionError()
     try:
         img = Image.open(BytesIO(buffer))
-        return np.array(img).reshape(shape)
+        arr = np.array(img)
+        if shape is not None:
+            arr = arr.reshape(shape)
+        return arr
     except Exception:
         raise SampleDecompressionError()
 
 
+def _get_bounding_shape(shapes: Sequence[Tuple[int]]) -> Tuple[int, int, int]:
+    if len(shapes) == 0:
+        return (0, 0, 0)
+    channels_shape = shapes[0][2:]
+    for shape in shapes:
+        if shape[2:] != channels_shape:
+            raise ValueError()
+    return (max(s[0] for s in shapes), sum(s[1] for s in shapes)) + channels_shape  # type: ignore
+
+
+def compress_multiple(arrays: Sequence[np.ndarray], compression: str) -> bytes:
+    """Compress multiple arrays of different shapes into a single buffer. Used for chunk wise compression."""
```

Comment:
is this the method that puts the samples onto 1 numpy array, and then does the compression? can you update the docstring to explain this a little more directly